docs/js/tiles.js

```diff
@@ -28,11 +28,28 @@ function submitQuery() {
     .map(toKeyValueString)
     .join("&")
 }
+function tesselace(){
+
+  // keep tesselace reference as long as tile definition is unchanged
+  var tesselace = ""
+  if (window.location.search.substr(1).includes("tesselace=")) {
+    // obtain tile definition from url
+    var urlTile = window.location.search.replace(/(.*[?&])?tile=/, "").replace(/&.*/, "")
+    // obtain tile definition from user interface
+    var configTile = TilesConfig(submitQuery()).tileDef
+    // compare
+    if (urlTile == configTile) {
+        // formulate tesselace reference
+        tesselace = window.location.search.replace(/(.*[?&])?tesselace=/, "tesselace=").replace(/&.*/, "&")
+    }
+  }
+  return tesselace
+}
 function showProto() {
 
   var config = TilesConfig(submitQuery())
   d3.select("#prototype").html(PrototypeDiagram.create(config))
-  d3.select("#link").node().href = "?" + submitQuery() // don't extract var, we might now have other form fields
+  d3.select("#link").node().href = "?" + tesselace() + submitQuery()// don't use config variable, form fields may have changed
```

Comment:
Now you are calling submitQuery 3 times. It looks a tiny function but browing the DOM might not be cheap, especially for large diagrams. Something in the following gist:

      var changedQuery = submitQuery()
      var config = TilesConfig(changedQuery)
      d3.select("#prototype").html(PrototypeDiagram.create(config))
      var newQuery = submitQuery() // calling again as the prototypeDiagram might have generated other form fields for stitches
      var oldQuery = window.location.search.substr(1)
      if (oldQuery.includes("tesselace=")) {
        var searchPrefix = /(.*[?&])?tile=/
        var searchSuffix = /&.*/
        var oldTile = findTileArgument(oldQuery)
        var newTile = findTileArgument(newQuery)
        ...
      }
      else d3.select("#link").node().href = "?" + newQuery